src/Components/List/List.jsx

```diff
@@ -38,18 +68,58 @@ const purchaseItem = (item, token) => {
         lastPurchased: new Date(),
         oldPurchased: item.lastPurchased,
         numberOfPurchases: numberOfPurchases,
-        calculatedEstimate: calculateEstimate(
-          item.frequency,
-          lastInterval,
-          numberOfPurchases,
-        ),
+        calculatedEstimate: calculatedEstimate,
       },
     });
 };
 
+//Color code based on days until next purchase or inactivity with corresponding aria label
+const colorCode = (item) => {
+  const daysSincePurchased = dayjs().diff(
+    getLastPurchaseDate(item, item.lastPurchased),
+    'day',
+  );
+  const estimatedCountdown = item.calculatedEstimate - daysSincePurchased;
+  const elapsedTime = dayjs().isSameOrAfter(
+    dayjs(getLastPurchaseDate(item, item.lastPurchased)).add(
+      item.calculatedEstimate * 2,
+      'day',
+    ),
+  );
+
+  if (elapsedTime === true) {
+    return ['inactive', 'Inactive'];
+  } else if (estimatedCountdown <= 7) {
+    return ['soon', 'Within 7 days'];
+  } else if (estimatedCountdown <= 14) {
+    return ['kindOfSoon', 'Between 7 and 14 days'];
+  } else if (14 < estimatedCountdown) {
+    return ['notSoon', 'More than 14 days'];
+  } else if (isNaN(estimatedCountdown)) {
+    return ['notBought', 'Never Bought Yet'];
+  } else {
+    return null;
+  }
+};
+
+// Sort items by soonest to latest estimated repurchase
+const sortItems = (a, b) => {
+  if (a.calculatedEstimate === b.calculatedEstimate) {
+    return a.name.localeCompare(b.name, 'en', { sensitivity: 'base' });
+  } else if (
+    a.calculatedEstimate < b.calculatedEstimate ||
+    b.calculatedEstimate === undefined
+  ) {
+    return -1;
+  } else {
+    return 1;
+  }
+};
+
 export default function List({ items, token }) {
   let history = useHistory();
 
+  // Once item is checked, it can't be rechecked for 24 hours and is disabled
```

Comment:
Thank you for adding comments to the functions! It's super helpful as our app grows.